packages/nars/src/Switch.re

```diff
@@ -30,13 +25,12 @@ let encoder =
         onValueChange:
           Some(
             ProtoEncoders.encodeCallback(
-              ~registerCallback=bridge.Instance.registerCallback,
-              ~callback=args => {
-                let callback = props##onValueChange;
-                callback(
-                  StructDecoders.(getFieldExn("value", args, getBool)),
-                );
-              },
+              ~bridge=bridge,
+              ~callback=
+                Callback.map(
+                  ~f=(struct_) => StructDecoders.(getValueField(getBool, struct_)),
```

Comment:
This convention with wrapping args in the struct_ is weird and just adds overhead.